synapse/replication/tcp/handler.py

```diff
@@ -82,6 +112,73 @@ def start_replication(self, hs):
         port = hs.config.worker_replication_port
         hs.get_reactor().connectTCP(host, port, self._factory)
 
+    async def on_REPLICATE(self, cmd: ReplicateCommand):
+        # We only want to announce positions by the writer of the streams.
+        # Currently this is just the master process.
+        if not self._is_master:
+            return
+
+        if not self._connections:
```

Comment:
is this not racy? the connection could drop between us getting a REPLICATE and processing it?